src/common-lib/distributions.ts

```diff
@@ -62,28 +41,23 @@ export const claimsUnwrappedAmount = ({
 
   if (circleDistClaimAmount) {
     const claimAmt = circleDistClaimAmount || 0;
-    const { fixedPaymentAmt, pricePerShare } = calc({
+    const fixedPaymentAmt = getUserClaimedFixedPaymentAmt({
       address,
       decimals: circleDistDecimals,
       fixedGifts: circleFixedGifts,
-      pricePerShare: circleDistPricePerShare,
     });
-    fixedPayment = fixedPaymentAmt * pricePerShare;
```

Comment:
@zashton i think i found a bug here? it was multiplying fixed payment amount by circle distribution's price per share. can you confirm that this was a bug, & if so, what do you think the impact was on users?

Reply:
hmmm should be correct, there are 4 cases here
combined dist (fixed + circle dist)
circle dist only .
fixed dist only
circle + fixed dist seperate distributions

that line handles the case of combined distribution where fixed+circle dist is using the same vault hence pricePerShare is the same. So if it is only a circle dist `fixedPaymentAmt` will be 0 because `circleFixedGifts` is empty and so `fixedPayment` will be also 0

the following `if (fixedGifts)` to handle when the distribution is only a fixed payment distribution .

Your update should have the same end result as the original code

Reply:
ah, of course, thanks for clearing that up!